Handling Facebook errors

Depado opened the pull request:
…ion calls

Also quick typo in the comments of the `Read` struct. 
I didn't include much detail in the error message (the code is already in the message anyway), and I have no idea what the `FbtraceID` is about ^^

Cheers

paked commented on response.go:
```diff
@@ -16,6 +16,21 @@ const (
+	Message   string `json:"message"`
+	Type      string `json:"type"`
+	Code      int    `json:"code"`
+	FbtraceID string `json:"fbtrace_id"`
```
Let's rename this field to `FBTraceID`

paked commented on response.go:
```diff
@@ -16,6 +16,21 @@ const (
 	SendMessageURL = "https://graph.facebook.com/v2.6/me/messages"
 )
 
+// QueryResponse is the response sent back by facebook when setting up things
```
Capitalise Facebook

paked commented on response.go:
```diff
@@ -16,6 +16,21 @@ const (
+	Result string      `json:"result,omitempty"`
+}
+
+// QueryError is representing an error sent back by facebook
```
Capitalise Facebook

paked commented:
Also, on another note: The commit you've sent doesn't seem to have an author associated with it. [I'd follow the instructions here](https://help.github.com/articles/set-up-git/#setting-up-git). Once you've done that, `git commit --amend && git push -f` should fix everything up!

depado commented:
Hello there ! Thanks for the feedback, I'll have a look at this today.

depado commented:
Hey there. 
So I made the changes you requested (comments and field name) and also added this behavior for the other functions that can potentially fail. 

Also I fixed my author thing I think. 

Does it look good to you ?

depado commented:
Sure thing :)

depado commented on messenger.go:
```diff
@@ -122,7 +122,11 @@ func (m *Messenger) ProfileByID(id int64) (Profile, error) {
 	resp, err := client.Do(req)
 	defer resp.Body.Close()
 
-	err = json.NewDecoder(resp.Body).Decode(&p)
```
That would break everything :smile:

paked replied:
😆

depado commented:
There's an issue with the `ProfileByID` function as the returned JSON is either the profile of the user or an error. (So it can either be a `Profile` or a `QueryResponse` type) I'll give that a try later. Thanks for the feedback :)

paked commented:
Alright. Good luck!

depado commented:
Hey there. Found a way to handle that kind of corner case. Could you review the changes ? 
It works pretty fine from what I can see (for example try to `ProfileByID(0)` it will return an error).

paked commented:
![https://media.giphy.com/media/6fxTfCp307Hnq/giphy.gif](https://media.giphy.com/media/6fxTfCp307Hnq/giphy.gif)

@Depado Thanks for your persistence on this PR! I'm very grateful 🎈
